### PR TITLE
133 poll in browser

### PR DESCRIPTION
**Pull request by lukasmatusiewicz**

### CODE REVIEW

File: src/main/resources/theme-resources/resources/pi-pollTransaction.worker.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        case 'url':
+            url = data.msg + "/validate/polltransaction";
+            break;
+        case 'transactionID':
+            params = "transaction_id=" + data.msg;
+            break;
+        case 'start':
+            if (url.length > 0 && params.length > 0)
+            {
+                setInterval("pollTransactionInBrowser()", 2000);
```
Comment (nilsbehlen) [inline src/main/resources/theme-resources/resources/pi-pollTransaction.worker.js:17]: 2s intervals are too high, we could do this every 300 or 500ms

File: src/main/resources/theme-resources/templates/privacyIDEA.ftl [hunk 2]
```diff
@@ -78,33 +81,89 @@
...
+                                {
                                     // Submit the form to pass the change to the authenticator
                                     document.getElementById("mode").value = newMode;
                                     document.getElementById("modeChanged").value = "true";
                                     document.forms["kc-otp-login-form"].submit();
                                 }
                             </script>
 
+                            <!-- Poll in browser section -->
+                            <#if transactionID?? && !(transactionID = "") && !(piServerUrl = "") && (pollInBrowserFailed = false)>
```
Comment (nilsbehlen) [inline src/main/resources/theme-resources/templates/privacyIDEA.ftl:100]: Please add some comment describing the rationale behind the condition

Comment (nilsbehlen) [reply src/main/resources/theme-resources/templates/privacyIDEA.ftl:100]: or generally how poll in browser (that section) fits in with the rest of the code

File: src/main/resources/theme/privacyidea/login/theme.properties [hunk 3]
```diff
@@ -1,3 +1,3 @@
 parent=base
 import=common/keycloak
-scripts=js/pi-webauthn.js
\ No newline at end of file
+scripts=js/pi-webauthn.js, js/pi-u2f.js, js/pi-pollTransaction.worker.js, js/pi-webWorkerUtils.js
```
Comment (nilsbehlen) [inline src/main/resources/theme/privacyidea/login/theme.properties:3]: js/pi-webWorkerUtils.js is not in this PR. Is the file not needed/removed?

File: src/main/resources/theme-resources/templates/privacyIDEA.ftl [hunk 4]
```diff
@@ -78,33 +81,89 @@
...
+                                <script>
+                                    window.onload = () =>
+                                    {
+                                        console.log("Trying to to poll in the browser...");
```
Comment (nilsbehlen) [inline src/main/resources/theme-resources/templates/privacyIDEA.ftl:104]: Please remove this. Only log if an error occurs

File: src/main/resources/theme-resources/templates/privacyIDEA.ftl [hunk 5]
```diff
@@ -78,33 +81,89 @@
...
+                                                worker.postMessage({'cmd': 'url', 'msg': '${piServerUrl}'});
+                                                worker.postMessage({'cmd': 'transactionID', 'msg': '${transactionID}'});
+                                                worker.postMessage({'cmd': 'start'});
+                                                worker.addEventListener('message', function (e)
+                                                {
+                                                    let data = e.data;
+                                                    switch (data.status)
+                                                    {
+                                                        case 'success':
+                                                            console.log(data.message);
```
Comment (nilsbehlen) [inline src/main/resources/theme-resources/templates/privacyIDEA.ftl:126]: remove pls, only log on error

**Review by nilsbehlen — CHANGES_REQUESTED** [review]
Please add an optional URL config option that will be used for poll in browser is set.

File: src/main/resources/theme-resources/resources/pi-pollTransaction.worker.js (hunk 1 above)
Comment (lukasmatusiewicz) [reply src/main/resources/theme-resources/resources/pi-pollTransaction.worker.js]: changed in [a1d037b](https://github.com/privacyidea/keycloak-provider/pull/136/commits/a1d037b7632e149529ad2480af03f01a88a1d8e5)

File: src/main/resources/theme/privacyidea/login/theme.properties (hunk 3 above)
Comment (lukasmatusiewicz) [reply src/main/resources/theme/privacyidea/login/theme.properties]: fixed in [e0018db](https://github.com/privacyidea/keycloak-provider/pull/136/commits/e0018db038754e4c330ee95d22f710397e64f239)

File: src/main/resources/theme-resources/templates/privacyIDEA.ftl (hunk 5 above)
Comment (lukasmatusiewicz) [reply src/main/resources/theme-resources/templates/privacyIDEA.ftl]: [7cbc8a1](https://github.com/privacyidea/keycloak-provider/pull/136/commits/7cbc8a17037a70277a2ed2dfb9a4d3246b4eecbf)

**Review by nilsbehlen — APPROVED** [review]

**Pull request merged**